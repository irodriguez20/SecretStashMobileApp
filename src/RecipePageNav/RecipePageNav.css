.RecipePageNav__back-button {
  margin: 45px 20px 25px auto;
}
.RecipePageNav__back-button:hover {
  background-color: #c8556a;
}

.RecipePageNav__folder-name {
  color: whitesmoke;
  font-family: var(--custom-font);
  font-size: 40px;
  height: 50vh;
  margin: 0 0 0 auto;
  text-orientation: upright;
  width: 75px;
  writing-mode: vertical-lr;
}

@media only screen and (max-width: 669px) {
  .RecipePageNav {
    display: flex;
    flex-direction: column;
    align-items: flex-end;
    margin-top: 85px;
  }
}
