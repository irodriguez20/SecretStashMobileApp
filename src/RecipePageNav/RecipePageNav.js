import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Context from '../Context'
import CircleButton from '../CircleButton/CircleButton'
import { findRecipe, findFolder } from '../recipes-helpers'
import './RecipePageNav.css'

class RecipePageNav extends Component {
    static defaultProps = {
        history: {
            goBack: () => { }
        },
        match: {
            params: {}
        }
    }

    static contextType = Context;

    render() {
        const { recipes, folders } = this.context
        const { recipeId } = this.props.match.params
        const recipe = findRecipe(recipes, recipeId) || {}
        const folder = findFolder(folders, recipe.folderid)
        return (
            <div className='RecipePageNav'>
                <CircleButton
                    tag='button'
                    role='link'
                    onClick={() => this.props.history.goBack()}
                    className='RecipePageNav__back-button'
                >
                    <FontAwesomeIcon icon='chevron-left' />
                    <br />
                    Back
                </CircleButton>
                {folder && (
                    <h3 className='RecipePageNav__folder-name'>
                        {folder.foldername}
                    </h3>
                )}
            </div>
        )
    }
}

export default RecipePageNav;