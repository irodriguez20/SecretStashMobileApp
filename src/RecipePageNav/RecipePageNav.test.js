import React from "react";
import RecipePageNav from "./RecipePageNav";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe(`RecipePageNav component`, () => {
    it("renders a .RecipePageNav by default", () => {
        const wrapper = shallow(<RecipePageNav />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    // enzyme doesn't support React.createContext
    it.skip("renders a h2 with folder name when in props", () => {
        const props = {
            match: {
                params: {
                    recipeId: "test-recipe-id"
                }
            }
        };
        const context = {
            recipes: [{ id: "test-recipe-id", folderid: "test-folder-id" }],
            folders: [{ id: "test-folder-id", name: "Important" }]
        };

        const h2 = shallow(<RecipePageNav {...props} />, context).find(
            ".RecipePageNav__folder-name"
        );
        expect(toJson(h2)).toMatchSnapshot();
    });
});
