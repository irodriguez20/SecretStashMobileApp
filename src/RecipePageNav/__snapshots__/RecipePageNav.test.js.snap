// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePageNav component renders a .RecipePageNav by default 1`] = `
<div
  className="RecipePageNav"
>
  <NavCircleButton
    className="RecipePageNav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavCircleButton>
</div>
`;
